vignettes/subsections/intro-customizations.Rmd

```diff
@@ -0,0 +1,221 @@
+From the high-level [`mmrm()`](https://openpharma.github.io/mmrm/main/reference/mmrm.html)
+interface, common changes to the default function call can be specified.
+
+## Control Function
+
+For fine control, [`mmrm_control()`](https://openpharma.github.io/mmrm/main/reference/mmrm_control.html)
+is provided. This function allows the user to choose the adjustment method for
+the degrees of freedom and the coefficients covariance matrix, specify optimization routines,
+number of cores to be used on Unix systems for trying several optimizers in parallel,
+provide a vector of starting parameter values, decide the action to be taken when the
+defined design matrix is singular, not drop unobserved visit levels.
+For example:
+
+```{r low-level-control, eval=FALSE}
+mmrm_control(
+  method = "Kenward-Roger",
+  optimizer = c("L-BFGS-B", "BFGS"),
+  n_cores = 2,
+  start = c(0, 1, 1, 0, 1, 0),
+  accept_singular = FALSE,
+  drop_visit_levels = FALSE
+)
+```
+
+Note that this control list can either be passed via the `control` argument
+to `mmrm`, or selected controls can be directly specified in the `mmrm` call.
+We will see this below.
+
+## REML or ML
+
+Users can specify if REML should be used (default) or if ML should be used in
+optimization.
+
+```{r common-changes-reml}
+fit_ml <- mmrm(
+  formula = FEV1 ~ RACE + ARMCD * AVISIT + us(AVISIT | USUBJID),
+  data = fev_data,
+  reml = FALSE
+)
+fit_ml
+```
+
+## Optimizer
```

Comment:
is it worth mentioning that we also support other custom optimizers?

Reply:
are there others in addition to this?

Users can specify which optimizer should be used, changing from the default of
four optimizers, which starts with `L-BFGS-B` and proceeds through the
other choices if optimization fails to converge. Other choices are `BFGS`, `CG`, and
`nlminb`.

`L-BFGS-B`, `BFGS` and `CG` are all implemented with `stats::optim()` and the Hessian
is not used, while `nlminb` is using `stats::nlminb()` which in turn uses both the
gradient and the Hessian (by default but can be switch off) for the optimization.